packages/gatsby-plugin-page-creator/src/path-utils.ts

```diff
@@ -51,37 +51,33 @@ const extractFieldWithoutUnionRegex = /\(.*\)__/g
 // {Model.bar} => bar
 // {Model.field__bar} => field__bar
 // {Model.field__(Union)__bar} => field__bar
-export function extractFieldWithoutUnion(filePart: string): string {
-  return (
-    extractField(filePart)
-      // Ignore union syntax
-      .replace(extractFieldWithoutUnionRegex, ``)
-  )
+export function extractFieldWithoutUnion(filePart: string): Array<string> {
```

Comment:
Maybe also update the comments above to indicate that it actually returns an array?